lib/vsc/kafka/cli.py

```diff
@@ -175,3 +180,88 @@ def do(self, dry_run):
 
         for resource, events in self.get_resource_events():
             self.produce(resource, events, dry_run)
+
+
+class ConsumerCLI(KafkaCLI):
+    """Consume data from kafka topics and prepare it as xdmod shred file input"""
+
+    CONSUMER_CLI_OPTIONS = {
+        'group': ("Kafka consumer group", None, "store", "xdmod"),
+        'timeout': ('Kafka consumer timeout in ms. If not set, loops forever', int, "store", None),
+    }
+
+    def make_options(self, defaults=None):
+        self.CLI_OPTIONS.update(self.CONSUMER_CLI_OPTIONS)
+        return super(ConsumerCLI, self).make_options(defaults=defaults)
+
+
+    def get_kafka_kwargs(self):
+        """Generate the kafka producer or consumer args"""
+
+        kwargs = super(ConsumerCLI, self).get_kafka_kwargs()
+
+        if self.options.timeout is not None:
+            kwargs["consumer_timeout_ms"] = self.options.timeout
+
+        # disable auto commit, so dry-run doesn't commit
+        kwargs.setdefault('enable_auto_commit', False)
+
+        return kwargs
+
+    def convert_msg(self, msg):
+        """
+        Process msg as JSON.
+        Return None on failure.
+        """
+        value = msg.value
+        if value:
+            try:
+                event = json.loads(value)
+            except ValueError:
```

Comment:
does IO error also give ValueError? Probably does not matter. it will fail hard if it does.